### DISCUSSION TITLE
Reporting tool not sufficient for Accounting / Old Invoice Export was removed

### DISCUSSION THREAD

**Discussion by ShopinBit** [btcpayserver/btcpayserver#5458] [Ideas / Feature Requests]
The new reporting tool is technically cool but it has two problems:

1. You can't add specific fields which, makes accounting super difficult. E.g. we need one of the additional fields called orderNumber which is the Ordernumber from our Shopsystem Shopware and is used by the new Shopware 6 Plugin by Coincharge (https://github.com/coincharge-io/CoinchargeBTCPayShopware)
We recommend adding a checkbox field where you can just click together all the fields you need for export so each shopowner can generate the best reports that he needs given local laws for bitcoin bookkeeping.

2. Other Coins seem to be ignored (specific case XMR)
No XMR transaction shows up in the new report. Might have to do with the implementation of it. Don't know where the report generates the data, but the XMR transactions are all visible in "Invoices"

For fast fix, you could reactivate the old export function for "Invoices" (that would be nice). We appreciate the time you spend here to work on this project.

Feel free to use voucher code BTCPAY in our store for 5€ off and to support the devs of BTCPay.

Best Regards
Lando

**Comment by Kukks** [comment]
> The new reporting tool is technically cool but it has two problems:
> 
> 1. You can't add specific fields which, makes accounting super difficult. E.g. we need one of the additional fields called orderNumber which is the Ordernumber from our Shopsystem Shopware and is used by the new Shopware 6 Plugin by Coincharge ([coincharge-io/CoinchargeBTCPayShopware](https://github.com/coincharge-io/CoinchargeBTCPayShopware))
>    We recommend adding a checkbox field where you can just click together all the fields you need for export so each shopowner can generate the best reports that he needs given local laws for bitcoin bookkeeping.
> 2. Other Coins seem to be ignored (specific case XMR)
>    No XMR transaction shows up in the new report. Might have to do with the implementation of it. Don't know where the report generates the data, but the XMR transactions are all visible in "Invoices"
> 
> For fast fix, you could reactivate the old export function for "Invoices" (that would be nice). We appreciate the time you spend here to work on this project.
> 
> Feel free to use voucher code BTCPAY in our store for 5€ off and to support the devs of BTCPay.
> 
> Best Regards Lando

1.) I am about to release a Dynamic Report plugin that allows you to create custom reports. It is a bit advanced in usage though.
2.) This was fixed in the latest release

**Reply by Kukks** [reply]
https://github.com/Kukks/BTCPayServerPlugins/tree/master/Plugins/BTCPayServer.Plugins.DynamicReports#readme

Done!
